### PR TITLE
SASS-2199 - Merge employment summary views.

### PR DESCRIPTION
**Pull request by georgievh**
### Description
Please include a brief summary of the change you've made and the issue that it fixes.

Add a link to the relevant story in Jira
[SASS-2199](https://jira.tools.tax.service.gov.uk/browse/SASS-2199)

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [x]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [x]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [x]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [x]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of main?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [x]  Have you checked the PR Builder passes?

### CODE REVIEW

**Review by ina-tu — CHANGES_REQUESTED** [review]
needs a rebase

File: it/controllers/employment/EmploymentSummaryControllerISpec.scala [hunk 1]
```diff
@@ -193,9 +200,164 @@ class MultipleEmploymentSummaryControllerISpec extends IntegrationTest with View
       val specific = user.specificExpectedResults.get
 
       s"language is ${welshTest(user.isWelsh)} and request is from an ${agentTest(user.isAgent)}" should {
+        "return the single employment summary EOY page" when {
+          "there is only one employment and its the EOY with expenses" which {
+            implicit lazy val result: WSResponse = {
+              authoriseAgentOrIndividual(user.isAgent)
+              userDataStub(anIncomeTaxUserData, nino, taxYearEOY)
+              urlGet(fullUrl(employmentSummaryUrl(taxYearEOY)), welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYear)))
+            }
```
[the remark this answers is not in the document]
Comment (georgievh) [reply it/controllers/employment/EmploymentSummaryControllerISpec.scala:209]: I only merged the tests that were already there, i.e. I did not add new or change existing ones. 
Updated as requested.

File: it/controllers/employment/EmploymentSummaryControllerISpec.scala [hunk 2]
```diff
@@ -193,9 +200,164 @@ class MultipleEmploymentSummaryControllerISpec extends IntegrationTest with View
...
+            linkCheck(s"$change$change $expenses", changeExpensesSelector, checkYourExpensesUrl(taxYearEOY))
+            linkCheck(s"$remove $remove $expenses", removeExpensesSelector, checkYourExpensesUrl(taxYearEOY))
+            buttonCheck(returnToOverview)
+          }
+
+          "there is only one employment and its the EOY without expenses" which {
+            implicit lazy val result: WSResponse = {
+              authoriseAgentOrIndividual(user.isAgent)
+              userDataStub(anIncomeTaxUserData.copy(employment = Some(anAllEmploymentData.copy(hmrcExpenses = None))), nino, taxYearEOY)
+              urlGet(fullUrl(employmentSummaryUrl(taxYearEOY)), welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYear)))
```
[the remark this answers is not in the document]
Comment (georgievh) [reply it/controllers/employment/EmploymentSummaryControllerISpec.scala]: Same as above

File: it/controllers/employment/EmploymentSummaryControllerISpec.scala [hunk 3]
```diff
@@ -363,5 +526,56 @@ class MultipleEmploymentSummaryControllerISpec extends IntegrationTest with View
...
+        result.header(HeaderNames.LOCATION).contains(overviewUrl(taxYear)) shouldBe true
+      }
+
+    }
+
+    "redirect the user to the Add Employment page when no data is in session EOY" which {
+      lazy val result: WSResponse = {
+        authoriseAgentOrIndividual(isAgent = true)
+        val employmentSources = Seq(EmploymentSource(employmentId = "001", employerName = "maggie", None, None, None, None, dateIgnored = Some("2020-03-11"), None, None, None))
+        userDataStub(IncomeTaxUserData(Some(anAllEmploymentData.copy(hmrcEmploymentData = employmentSources))), nino, taxYear - 1)
```
[the remark this answers is not in the document]
Comment (georgievh) [reply it/controllers/employment/EmploymentSummaryControllerISpec.scala]: Same as above

File: it/controllers/employment/EmploymentSummaryControllerISpec.scala [hunk 4]
```diff
@@ -363,5 +526,56 @@ class MultipleEmploymentSummaryControllerISpec extends IntegrationTest with View
...
+        urlGet(s"$appUrl/$taxYearEOY/employment-summary", follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYear - 1)))
```
[the remark this answers is not in the document]
Comment (georgievh) [reply it/controllers/employment/EmploymentSummaryControllerISpec.scala]: Same as above

**Review by ina-tu — APPROVED** [review]

**Pull request merged**